src/kernels/helpers.ts

```diff
@@ -486,6 +486,11 @@ export function compareKernels(
             return 1;
         } else if (b === activeInterpreterConnection) {
             return -1;
+            // Interpreter should trump kernelspec
+        } else if (a.kind === 'startUsingPythonInterpreter' && b.kind !== 'startUsingPythonInterpreter') {
```

Comment:
Out of interest, why would this be the case?

Reply:
Well the tests were guaranteeing that so I assume that was the wanted behavior. I think before it was just luck that it was happening.

Reply:
In the test that was failing it would get to this point and the two kernelspecs would be equivalent (start from kernelspec and start from interpreter). There was nothing that would distinguish them.